sdk/metric/metricdata/metricdatatest/assertion.go

```diff
@@ -53,6 +52,14 @@ type Datatypes interface {
 	// Aggregation and Value type from metricdata are not included here.
 }
 
+// TB is an interface common to `testing.T` and `testing.B` but without the
+// private method of `testing.TB`, so other testing packages can rely on it as
+// well.
+type TB interface {
```

Comment:
I think we should add a warning like here https://github.com/open-telemetry/opentelemetry-go/pull/4396

Also I think we should add:

```
// Warning: methods may be added to this interface in minor releases.
```